upf/upf_api.c

```diff
@@ -477,23 +484,19 @@ vl_api_upf_nwi_add_del_t_handler (vl_api_upf_nwi_add_del_t * mp)
 {
   vl_api_upf_nwi_add_del_reply_t *rmp = NULL;
   upf_main_t *sm = &upf_main;
-  u8 *name = 0;
   u8 *nwi_name = 0;
   u32 ip4_table_id = 0;
   u32 ip6_table_id = 0;
   int rv = 0;
 
-  /* Make sure name is null terminated */
-  mp->name[sizeof (mp->name) - 1] = 0;
-  name = format (0, "%s", mp->name);
-  nwi_name = upf_name_to_labels (name);
-  vec_free (name);
+  vec_validate (nwi_name, mp->nwi_len - 1);
```

Comment:
vec_validate (vec, idx) stands to check if vector element exists for given index. If idx > current length, it will resize the vector so idx will be the last index accessible. So, the length of the vector after resizing will be idx + 1.